### ISSUE TITLE
Support Lemonade extensions for attachment forwarding without download on IMAP

### ISSUE THREAD

**Issue by Hooloovoo** [k9mail/k-9#1361]
### Expected behaviour
If an email message in a K9 IMAP account contains attachments, a user should be able to forward the email and its attachments without downloading the attachments. 
See also https://code.google.com/archive/p/k9mail/issues/216

### Actual behaviour
The user has two options: forward the email without attachments; or download the attachments before forwarding the email.

Specifically, this requires the implementation of three Lemonade extensions: CATENATE, URLAUTH and BURL.

I believe that Isode's and Oracle's email servers already support the server side of these. Dovecot supports two and is scheduled to support the final one in its next release (2.3; https://www.ietf.org/mail-archive/web/apps-discuss/current/msg13450.html ), which should make this far more ubiquitous.

Trojita (and therefore Ubuntu Touch's email client) supports this feature:
https://github.com/benkaiser/trojita/blob/master/docs/masters/mobile-imap.tex

Other useful background on this can be found in the Mail in a Box issue about this:
https://github.com/mail-in-a-box/mailinabox/issues/801

**Comment by stephanbosch** [comment]
Dovecot has released v2.3 with the new submission server that includes BURL support. In the v2.3.0 technology preview it was still quite buggy, but v2.3.1 will be released in a few weeks, which should address all of these issues.

For development, you can of course already install Dovecot git master.

**Comment by evert-mouw** [comment]
This would be a very nice feature to have, not only to limit network traffic, but it might also speed up the sending of messages with attachments.

**Comment by mbethke** [comment]
Indeed, this would speed up sending *any* message about two-fold as it doesn't have to be uploaded both via SMTP and IMAP. Considering Dovecot's market share, this would probably benefit a majority of users right away.
